code/game/gamemodes/objective.dm

```diff
@@ -567,16 +567,22 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 /datum/objective/steal/proc/select_target()
 	var/list/possible_items_all = GLOB.potential_theft_objectives+"custom"
 	var/new_target = input("Select target:", "Objective target", null) as null|anything in possible_items_all
-	if(!new_target) return
+	if(!new_target)
+		return
+
 	if(new_target == "custom")
 		var/datum/theft_objective/O=new
```

Comment:
While you're here, this O needs to be qdeleted for the early returns on 576 and 583